runtime/lua/vim/filetype/detect.lua

```diff
@@ -257,11 +256,13 @@ end
 -- This function checks if one of the first five lines start with a dot. In
 -- that case it is probably an nroff file.
 function M.nroff(bufnr)
-  for _, line in iter_lines(bufnr, 1, 5) do
-    if line:find("^%.") then
+  for _, line in ipairs(get_lines(bufnr, 1, 5)) do
+    if line:sub(1, 1) == "." then
```

Comment:
Why did you switch back from `find` to `sub`?

Reply:
I thought it was more efficient in this case since we only need to check the first character. Do you prefer to use `find` for everything? I was only aware of the preference of `find` vs. `match`.

Reply:
Yes, I think so -- unlike `sub`, `find` doesn't create a new string. 

In any case, I think we should be consistent for now; we can still switch en bloc later if benchmarks show that `sub` is in fact faster.